cypress/e2e/safe-apps/permissions_settings.cy.js

```diff
@@ -0,0 +1,146 @@
+import { BROWSER_PERMISSIONS_KEY, SAFE_PERMISSIONS_KEY } from './constants'
+import { TEST_SAFE } from './constants'
+
+let $dapps = []
+
+describe('The Safe Apps permissions settings section', () => {
+  before(() => {
+    cy.on('window:before:load', (window) => {
+      window.localStorage.setItem(
+        BROWSER_PERMISSIONS_KEY,
+        JSON.stringify({
+          'https://app1.com': [
+            { feature: 'camera', status: 'granted' },
+            { feature: 'fullscreen', status: 'granted' },
+            { feature: 'geolocation', status: 'granted' },
+          ],
+          'https://app2.com': [{ feature: 'microphone', status: 'granted' }],
+          'https://app3.com': [{ feature: 'camera', status: 'denied' }],
+        }),
+      )
+      window.localStorage.setItem(
+        SAFE_PERMISSIONS_KEY,
+        JSON.stringify({
+          'https://app2.com': [
+            {
+              invoker: 'https://app1.com',
+              parentCapability: 'requestAddressBook',
+              date: 1666103778276,
+              caveats: [],
+            },
+          ],
+          'https://app4.com': [
+            {
+              invoker: 'https://app3.com',
+              parentCapability: 'requestAddressBook',
+              date: 1666103787026,
+              caveats: [],
+            },
+          ],
+        }),
+      )
+    })
+
+    cy.visit(`${TEST_SAFE}/settings/safe-apps`, { failOnStatusCode: false })
+  })
+
+  it('should show the dApps with allowed permissions', () => {
+    cy.get('main .MuiPaper-root > .MuiGrid-item')
```

Comment:
Not a big fan of getting hooked on implementation details. Is there a different approach to getting these than depending on Mui classes?

Reply:
I strongly agree but couldn't figure out because the web team don't want to add test ids or ids I think. At least in safe-react

Perhaps an article tag?

Reply:
Can't you target the title and traverse elements from there?

Reply:
>  don't want to add test ids or ids I think

Why? I agree it should be the last resort, but we shouldn't ban them

Reply:
i don't get the difference, i'm going to query directly for the app and then going to the parent, sibling ... etc At the end going to implementation details isn't it?